consensus/src/experimental/commit_phase.rs

```diff
@@ -376,6 +417,13 @@ impl CommitPhase {
                             }
                         }
                     }
+                    // callback event might come when self.blocks is not empty
+                    reset_event_callback = self.reset_event_rx.select_next_some() => {
```

Comment:
this seems hard to use a single channel though

Reply:
yeah